draft-ietf-openpgp-pqc.md

```diff
@@ -577,62 +540,55 @@ The ML-KEM + ECDH composite public-key encryption schemes are built according to
 For the composite KEM schemes defined in {{kem-alg-specs}} the following procedure, justified in {{sec-fixed-info}}, MUST be used to derive a string to use as binding between the KEK and the communication parties.
 
     //   Input:
-    //   algID     - the algorithm ID encoded as octet
+    //   algID - the algorithm ID encoded as octet
+    //
+    //   Constants:
+    //   domSeparation - the UTF-8 encoding of the string
+    //                   "OpenPGPCompositeKeyDerivationFunction"
+
+    fixedInfo = algID || domSeparation
```

Comment:
I also think we should encode it at the end, if we do so.

On the other hand, in this (unlinkely) case, I think we could also alter the domain separation? We'll need a new key/PKESK/... version for that

Reply:
> When viewing both the length, and the KEM input as "arbitrary length and arbitrary content" strings, wouldn't we have the same problem even when encoding the length of the algorithm ID? Since we have multiple variable-length fields following each other without a clear boundary. Thus, we need to prefix the length of the other variable input (the KEM data), too.

What I propose aims exactly at solving this problem. When string is anambiguously parsable from the rear, collisions cannot happen: after the algorithm ID has been parsed from the rear, the lengths of the other fields are known.

Changing the domain separation string might again lead to possible collisions. Thus the new  domain separation string would need to be of the same length as the current one. Maybe we should include a two digit (i.e. two byte) version number in the `domSeparation` string and set it to "00" for now.

Reply:
Making the domain separation `OpenPGPCompositeKDFv1`